### PR TITLE
Adding functions to preconfigure the FXP

### PR DESCRIPTION
**Pull request by palade**
Workaround added to preconfigure the FXP.

### CODE REVIEW

File: ipu-plugin/pkg/ipuplugin/lifecycleservice.go [hunk 1]
```diff
@@ -485,6 +485,82 @@ func (e *ExecutableHandlerImpl) validate() bool {
 	return true
 }
 
+func executeScript(script string) {
+	var stdout bytes.Buffer
+	var stderr bytes.Buffer
+	// Execute the shell script
+
+	cmd := exec.Command("bash", "-c", script)
```
Comment (ahalimx86) [inline ipu-plugin/pkg/ipuplugin/lifecycleservice.go:493]: Will this work without bash? when we run this from container bash will not be available by default.

Comment (palade) [reply ipu-plugin/pkg/ipuplugin/lifecycleservice.go]: Changed this to sh

File: ipu-plugin/pkg/ipuplugin/lifecycleservice.go [hunk 2]
```diff
@@ -485,6 +485,82 @@ func (e *ExecutableHandlerImpl) validate() bool {
...
+	executeScript(`
+IFS=$'\n' read -r -d '' -a vf_mac_addresses < <(ssh -o StrictHostKeyChecking=no -o ConnectTimeout=10 root@192.168.0.1 "/usr/bin/cli_client -cq" \
+    | awk '{if(($4 == "0x0") && ($6 == "yes")) {print $17}}')
+
+for src_mac in "${vf_mac_addresses[@]}"; do
+    for dst_mac in "${vf_mac_addresses[@]}"; do
+        smac=$(echo "$src_mac" | tr -d ':')
+        if [ "$src_mac" != "$dst_mac" ]; then
+            dmac=$(echo "$dst_mac" | tr -d ':')
+            vsi=$(echo "$src_mac" | awk -F: '{print $2}')
```
Comment (ahalimx86) [inline ipu-plugin/pkg/ipuplugin/lifecycleservice.go:527]: I think it's good idea to install gawk in the Dockerfile for this awk to work in container environment

Comment (palade) [reply ipu-plugin/pkg/ipuplugin/lifecycleservice.go]: Awk removed - handling this in go

File: ipu-plugin/pkg/ipuplugin/lifecycleservice.go [hunk 3]
```diff
@@ -487,6 +489,104 @@
...
+				vsi := strings.Split(vfMacList[i], ":")[1]
+				targetVsi := strings.Split(vfMacList[j], ":")[1]
+				vf2Port, err := strconv.ParseInt(targetVsi, 16, 64)
+
+				if err != nil {
+					fmt.Printf("error converting hex to decimal: %v", err)
+					return
+				}
+
+				vf2Port += 16
```
[the remark this answers is not in the document]
Comment (palade) [reply ipu-plugin/pkg/ipuplugin/lifecycleservice.go]: Removed

File: ipu-plugin/pkg/ipuplugin/lifecycleservice.go [hunk 4]
```diff
@@ -487,6 +489,102 @@ func (e *ExecutableHandlerImpl) validate() bool {
...
+		| awk '{if(($4 == "0x0") && ($6 == "yes")) {print $17}}'`)
+
+	if err != nil {
+		fmt.Printf("Unable to reach the IMC %v", err)
+		return
+	}
+
+	vfMacList := strings.Split(strings.TrimSpace(output), "\n")
+
+	for i := range vfMacList {
```
Comment (ahalimx86) [inline ipu-plugin/pkg/ipuplugin/lifecycleservice.go:566]: Can this for loop for parsing the mac and vsi info put into a function and then re-use for both in AddFxpRules and DelFxpRules? It looks quite similar other than the actual utils.RunP4rtCtlCommand.

Comment (palade) [reply ipu-plugin/pkg/ipuplugin/lifecycleservice.go:566]: This code will be refactored once we add the code to generate the rules for network function comms. Will address  this comment in that PR.

**Review by ahalimx86 — APPROVED** [review]

**Pull request merged**